bingo.py

```diff
@@ -70,8 +81,19 @@ def print(self):
 
         print('')
 
-
-
+    def score(self):
+        free_ids = [x[0] * self.w + x[1] for x in self.free]
+        checked_str = [
+            1 if self.linear_grid[i] in self.chosen_o or i in free_ids else 0
+            for i in range(self.h * self.w)
+        ]
+        found = []
+        for temp_name, template in TEMPLATES.items():
+            temp_list = temp2list(template)
+            if check(checked_str, temp_list):
+                found.append(temp_name)
+        found.sort()
```

Comment:
If you sort ahead of time by name, then don't need this call.

But my comments are for easy readability, and your code has been written for optimization 🙃

Reply:
Sorry, didn't get this part. The main reason I have a sort is to fix the order of displayed matches. An ordered dictionary for templates will also do the trick and avoid the sorting.